pkg/wallet/wallet.go

```diff
@@ -63,12 +68,71 @@ func (w *BaseWallet) DecryptMessage(encMessage []byte, toVerKey string) ([]byte,
 
 // PackMessage Pack a message for one or more recipients.
 func (w *BaseWallet) PackMessage(envelope *Envelope) ([]byte, error) {
-	return nil, fmt.Errorf("not implemented")
+	if envelope == nil {
+		return nil, errors.New("envelope argument is nil")
+	}
+	// get keypair from db
+	key, err := w.getKey(envelope.FromVerKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed from getKey: %w", err)
+	}
+	// decode base58 public key
+	pubBytes := base58.Decode(key.Public)
+
+	// create 32 byte key
+	var pubKey [chacha.KeySize]byte
+	copy(pubKey[:], pubBytes)
+
+	var recipients []*[chacha.KeySize]byte
+	for _, verKey := range envelope.ToVerKeys {
+		// decode base58 ver key
+		verKeyBytes := base58.Decode(verKey)
+		// create 32 byte key
+		var recipient [chacha.KeySize]byte
+		copy(recipient[:], verKeyBytes)
+		recipients = append(recipients, &recipient)
+	}
+
+	// encrypt message
+	bytes, err := w.crypter.Encrypt(envelope.Message,
+		crypto.KeyPair{Priv: key.Private, Pub: &pubKey}, recipients)
```

Comment:
I don't understand why private key is used as is while public key had to be base58 decoded first. Shouldn't private key be base58 decoded too? Consistency is key.

what format do we expect to store keys in a wallet?

Reply:
done